### PR TITLE
ITE: soc: Add the variants of it82xx2bw

### PR DESCRIPTION
**Pull request by GTLin08**
This PR includes five commits, 
1. Using the SOC_IT8XXX2_REG_SET_V2 instead of constantly adding new variants of the IT82XX2 SoC.
2. Remove underscore makes config name consistent.
3. Add the description to migration-guide-3.7 and release-notes-3.7
4. Adding it82202bw variant.
5. Adding it82302bw variant.

### CODE REVIEW

File: soc/ite/ec/it8xxx2/Kconfig.soc [hunk 1]
```diff
@@ -50,3 +54,4 @@ config SOC
 	default "it82002aw" if SOC_IT82002_AW
 	default "it82202ax" if SOC_IT82202_AX
 	default "it82302ax" if SOC_IT82302_AX
+	default "it82202bw" if SOC_IT82202_BW
```
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:61]: I did not notice before, why is there an underscore in these Kconfig symbols? The names here should match e.g. `default "it82202_bw" if SOC_IT82202_BW` or `default "it82202bw" if SOC_IT82202BW`

File: soc/ite/ec/it8xxx2/Kconfig.soc [hunk 2]
```diff
@@ -39,6 +39,10 @@ config SOC_IT82002_AW
 	bool
 	select SOC_IT8XXX2
 
+config SOC_IT82202_BW
+	bool
+	select SOC_IT8XXX2
+
 config SOC_SERIES
 	default "it8xxx2" if SOC_SERIES_ITE_IT8XXX2
```
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:51]: Same with this, the names should match e.g. `default "it8xxx2" if SOC_SERIES_IT8XXX2` or `default "ite_it8xxx2" if SOC_SERIES_ITE_IT8XXX2`

File: soc/ite/ec/it8xxx2/Kconfig.soc (hunk 1 above)
Comment (GTLin08) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: Remove underscore.

File: soc/ite/ec/it8xxx2/Kconfig.soc (hunk 2 above)
Comment (GTLin08) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: Remove "_ITE".

**Review by Dino-Li — DISMISSED** [review]

File: boards/ite/it82xx2_evb/Kconfig.it82xx2_evb [hunk 3]
```diff
@@ -2,4 +2,4 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_IT82XX2_EVB
-	select SOC_IT82202_AX
+	select SOC_IT82202AX
```
Comment (keith-zephyr) [inline boards/ite/it82xx2_evb/Kconfig.it82xx2_evb:5]: You need to document this change in the release notes and migration guide.

Comment (GTLin08) [reply boards/ite/it82xx2_evb/Kconfig.it82xx2_evb:5]: Done.

File: doc/releases/migration-guide-3.7.rst [hunk 4]
```diff
@@ -30,6 +30,10 @@ Boards
 
 * Reordered D1 and D0 in the `pro_micro` connector gpio-map for SparkFun Pro Micro RP2040 to match
   original Pro Micro definition. Out-of-tree shields must be updated to reflect this change.
+* ITE: Rename all SoC variant names, e.g., ``SOC_IT82202_AX`` is renamed to ``SOC_IT82202AX``.
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:33]: Should this say?
```suggestion
* ITE: Rename all SoC variant names kconfig options, e.g., ``CONFIG_SOC_IT82202_AX`` is renamed to ``CONFIG_SOC_IT82202AX``.
```
if it is only the kconfig options and nothing else that changes (i.e. only those who have custom boards or happen to use those kconfig options in their code may realize)

Comment (GTLin08) [reply doc/releases/migration-guide-3.7.rst]: Done.

**Review by aescolar — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by Dino-Li — APPROVED** [review]

**Pull request merged**